File: swift/sort/bubble_sort.swift
func bubble_sort(target: inout [Int]) {
    for i in 0..<target.count {
        for j in 0..<target.count - i - 1 {
            if target[j] > target[j + 1] {
                let temp = target[j]
                target[j] = target[j + 1]
                target[j + 1] = temp
            }
        }
    }
}

File: elixir/sort/bubble_sort.ex
defmodule BubbleSort do
  def sort(list) when is_list(list) do
    do_sort(list, length(list))
  end

  defp do_sort(list, 0), do: list

  defp do_sort(list, count) do
    new_list = bubble_pass(list)
    do_sort(new_list, count - 1)
  end

  defp bubble_pass([a, b | rest]) when a > b do
    [b | bubble_pass([a | rest])]
  end
  defp bubble_pass([a, b | rest]) do
    [a | bubble_pass([b | rest])]
  end
  defp bubble_pass(list), do: list
end
